File: l-umi/Week_26/BOJ_15654_N과M_5.py
from itertools import permutations

def generate_sequences(N, M, numbers):
    # 주어진 숫자들 중에서 M개를 선택하는 모든 순열을 생성
    sequences = permutations(sorted(numbers), M)
    
    # 각 순열을 정렬된 순서로 출력
    for seq in sequences:
        print(' '.join(map(str, seq)))

# 입력 받기
N, M = map(int, input().split())
numbers = list(map(int, input().split()))

# 수열 생성 및 출력
generate_sequences(N, M, numbers)

File: yunva17/Week_26/BOJ_15654_N과M5/BOJ_15654_N과M5.js
let input = require("fs")
  // .readFileSync('/dev/stdin')
  .readFileSync("Week_26/BOJ_15654_N과M5/input.txt")
  .toString()
  .trim()
  .split("\n");

const [N, M] = input[0].split(" ").map(Number);
const Nlist = input[1]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

let numbers = [];
let isVisited = Array.from({ length: N }, () => false);

const solve = (count) => {
  if (count === M) {
    console.log(numbers.join(" "));

    return;
  }

  for (let i = 0; i < N; i++) {
    if (isVisited[i]) continue; // 중복 패스

    numbers[count] = Nlist[i];
    isVisited[i] = true;
    solve(count + 1);
    isVisited[i] = false;
  }
};

solve(0);
